src/aquarium-optimized/Aquarium.cpp

```diff
@@ -157,42 +157,62 @@ BACKENDTYPE Aquarium::getBackendType(const std::string &backendPath)
 
 bool Aquarium::init(int argc, char **argv)
 {
-    mFactory = new ContextFactory();
+    std::string backend;
+    int windowWidth = 0;
+    int windowHeight = 0;
 
-    char *pNext;
-    for (int i = 1; i < argc; ++i)
+    cxxopts::Options options(argv[0], "A native implementation of WebGL Aquarium");
+    options.allow_unrecognised_options().add_options()
+        ("help", "Print help")
+        ("backend", "Specifies running a certain backend, 'opengl', 'dawn_d3d12', 'dawn_vulkan', 'dawn_metal', 'dawn_opengl', 'angle', 'd3d12'", cxxopts::value<std::string>())
+        ("alpha-blending", "Format is [0-1]|false. Force enable alpha blending to a specific value or disable alpha blending for all models. By default, alpha blending is enabled.", cxxopts::value<std::string>())
+        ("buffer-mapping-async", "Upload uniforms by buffer mapping async for Dawn backend")
+        ("disable-control-panel", "Turn off control panel. You can show fps by passing '--print-log --test-time 30' to print the fps to cmd line.")
+        ("disable-d3d12-render-pass", "Turn off render pass for dawn_d3d12 and d3d12 backend")
+        ("disable-dawn-validation", "Turn off dawn validation")
+        ("disable-dynamic-buffer-offset", "The path is to test individual draw by creating many binding groups on dawn backend. By default, dynamic buffer offset is enabled. This option is only supported on dawn backend.")
+        ("discrete-gpu", "Choose discrete gpu to render the application. This is only supported on Dawn and D3D12 backend.")
+        ("integrated-gpu", "Choose integrated gpu to render the application. This is only supported on Dawn and D3D12 backend.")
+        ("enable-full-screen-mode", "Render aquarium in full screen mode instead of window mode")
+        ("msaa-sample-count", "specifies sample count of MSAA, while 1 for non-MSAA", cxxopts::value<int>())
+        ("num-fish", "specifies how many fishes will be rendered.", cxxopts::value<int>(mCurFishCount)->default_value("1000"))
+        ("print-log", "print logs including avarage fps when exit the application.")
+        ("simulating-fish-come-and-go", "Load fish behavior from FishBehavior.json. The mode is only implemented for Dawn backend.")
+        ("test-time", "Render the application for some seconds and then exit, and the application will run 5 min by default.", cxxopts::value<int>(mTestTime))
+        ("turn-off-vsync", "Unlimit 60 fps")
+        ("window-size", "Input window size with format [width],[height]", cxxopts::value<std::string>())
+        ;
```

Comment:
The dangling semicolon does not fully look good to me (and maybe clang-format). I would propose such a change:

```suggestion
        ("help", "Print help"); // Put "help" in the last. It's unlikely the line will be touched in any future option changes.
```

Reply:
done